### ISSUE TITLE
FilePermissionError when installing

### ISSUE THREAD

**Issue by seventhaxis** [donmelton/video_transcoding#321]
Fresh install of macOS and Homebrew, but getting an error when attempting the following command:

```
$ sudo gem install video_transcoding
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /usr/bin directory.
```
Software | Version | Notes
-|-|-
macOS | 10.15.5 | 
Homebrew | 2.3.0 | `brew --version`
Ruby (preinstalled) | 2.6.3p62 | `ruby --version`
Ruby | 2.7.1_2 | installed with `brew install ruby`

```
$ which ruby
/usr/bin/ruby
```

```
$ gem environment
RubyGems Environment:
  - RUBYGEMS VERSION: 3.1.4
  - RUBY VERSION: 2.6.3 (2019-04-16 patchlevel 62) [universal.x86_64-darwin19]
  - INSTALLATION DIRECTORY: /Library/Ruby/Gems/2.6.0
  - USER INSTALLATION DIRECTORY: /Users/seventhaxis/.gem/ruby/2.6.0
  - RUBY EXECUTABLE: /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
  - GIT EXECUTABLE: /usr/bin/git
  - EXECUTABLE DIRECTORY: /usr/bin
  - SPEC CACHE DIRECTORY: /Users/seventhaxis/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /Library/Ruby/Site
  - RUBYGEMS PLATFORMS:
    - ruby
    - universal-darwin-19
  - GEM PATHS:
     - /Library/Ruby/Gems/2.6.0
     - /Users/seventhaxis/.gem/ruby/2.6.0
     - /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/gems/2.6.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /usr/local/bin
     - /usr/bin
     - /bin
     - /usr/sbin
     - /sbin
     - /Library/Apple/usr/bin
```

Help please...

![](https://media.giphy.com/media/9Y5BbDSkSTiY8/giphy.gif)

**Edit** Additionally, I'm getting a _Permission denied_ error when executing the following command:

```
$ brew install ruby
warning: unable to access '/Users/seventhaxis/.config/git/attributes': Permission denied
```

**Comment by donmelton** [comment]
@seventhaxis I'm sorry you're having this problem.

I would not recommend installing Gems into the system paths on macOS. And I have no idea why that's not working for you either. The system-installed Ruby is a mess.

So let's see if we can fix your Homebrew installation. That said, I would recommend visiting that Homebrew website and support group because that can also be tricky.

Are you able to install anything else with Homebrew?

**Comment by seventhaxis** [comment]
Thanks Don -- I just performed a _new_ fresh install of macOS 10.15.5. Just got some basics going for now but yes I was able to install a bunch of items from Homebrew without issue. What would you suggest the process be for installing your tools?

Just so you're aware, I already have `ffmpeg`, `mkvtoolnix`, `mp4v2`, and `mpv` installed. Plus I manually installed Handbrake-CLI (because of issues I ran into with Handbrake last time I set this all up).

**Comment by seventhaxis** [comment]
Just to update, I tried installing other gems with the above environment variables incl. the default RubyGems v3.0.3 and they worked fine. So I tried video_transcoding once again, and it seemed to install without issue. I have no idea what went wrong except that I updated to the latest RubyGems   v3.1.4 yesterday afternoon before trying to install your package.

Thanks for the response and help as always!
